tests/Unit/TestHelpers.hpp

```diff
@@ -91,6 +91,8 @@ T serialize_and_deserialize(const T& t) {
 }
 
 /// Test for copy semantics assuming operator== is implement correctly
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```

Comment:
Alright, so we still don't need to `push` the warnings for the entire function is what I get from that.